common/assets/javascripts/analytics.js

```diff
@@ -0,0 +1,42 @@
+const { nodeListForEach } = require('./utils')
+
+function Analytics($module) {
+  this.cacheEls($module)
+}
+
+Analytics.prototype = {
+  init: function() {
+    this.cacheEls()
+    this.render()
+  },
+
+  cacheEls: function() {
+    this.$errors = document.querySelectorAll('.govuk-error-summary__list li a')
+  },
+
+  render: function() {
+    nodeListForEach(this.$errors, this.trackError.bind(this))
+  },
+
+  trackError: function(error) {
+    const fieldId = error.href.substring(error.href.indexOf('#') + 1)
+    const errorMessage = error.text
+
+    this.sendEvent({
+      action: fieldId,
+      category: 'validation_error',
+      label: errorMessage,
+    })
+  },
+
+  sendEvent: function({ action, category, label }) {
+    try {
```

Comment:
how come this is in a `try/catch`? its just pushing into the `dataLayer`. Are you expecting this to error?

Reply:
This is so that this function doesn't error when `gtag` is not available on the window. For example when a Google Analytics ID hasn't been set as an environment variable and GA isn't available.